undofx/src/main/java/org/fxmisc/undo/impl/UndoManagerImpl.java

```diff
@@ -209,7 +229,7 @@ private void addChange(C change) {
     @SuppressWarnings("unchecked")
     private C[] merge(C c1, C c2) {
         Optional<C> merged = merge.apply(c1, c2);
-        if(merged.isPresent()) {
+        if(merged.isPresent() && isNotIdentity.test(merged.get())) {
             return (C[]) new Object[] { merged.get() };
         } else {
             return (C[]) new Object[] { c1, c2 };
```

Comment:
> A single boolean flag (canMerge) will not suffice, since we can no longer forget that the user called preventMerge once the next change (c2 in my previous comment) arrives. 

Mm... true. I assumed the change annihilation would occur every time `preventMerge` is called, but that doesn't happen every time.

> One way to store in the queue Either<Barrier, C>, where left case means to prevent merge of changes to the left and right of the "barrier".

Not sure where/how `Either<Barrier, C>` would be used. I need a bit more context to connect the dots here.

One question I have: when a user calls `preventMerge`, is the previous change in the queue (the next undo) supposed to never be merged with any future changes? Given the following timeline where `|` represents the temporary barrier and `~` represents the current position in the queue:
````
.... c1 |~ // prevent merge called
.... c1 c2 c3 ~ // more changes pushed
.... c1 ~ c2 c3 // undo called twice
````
When a new change (`c4`) is emitted, should `c1` merge with `c4`? It would seem the latest release would merge them. Is that supposed to happen?

If not, perhaps we can make the following changes:
- The type in `List<C> changes` becomes `List<Either<C, C>> changes`
- the left side would represent an unmergeable change
- the right side would represent a mergeable change
- when `preventMerge` is called, we'd call 
````
if (currentPosition > 0) {
    Either<C, C> either = changes.get(currentPosition - 1);
    if (either.isRight()) {
        C c = either.getRight();
        changes.set(currentPosition - 1, Either.left(c));
        nextIsUnmergeable = false;
    }
} else {
    nextIsUnmergeable = true;
}
````
- when a change is pushed and `nextIsUnmergeable` is true, it stores the change as left and not as right.